src/meta/MetaServiceUtils.cpp

```diff
@@ -422,9 +422,69 @@ std::string MetaServiceUtils::assembleSegmentKey(const std::string& segment,
+                                                nebula::cpp2::SchemaProp&  prop,
+                                                std::vector<kvstore::KV>& defaultKVs,
+                                                std::vector<std::string>& removeDefaultKeys,
+                                                const GraphSpaceID spaceId,
+                                                const TagID Id,
+                                                const nebula::cpp2::ColumnDef col,
+                                                const cpp2::AlterSchemaOp op) {
+    static_assert(std::is_same<TagID, EdgeType>::value, "");
+    auto name = col.get_name();
+    auto dKey = MetaServiceUtils::defaultKey(spaceId, Id, name);
+    std::string defaultValue;
+    if (col.__isset.default_value) {
+        auto value = col.get_default_value();
+        switch (col.get_type().get_type()) {
+            case nebula::cpp2::SupportedType::BOOL:
+                if (value->getType() != nebula::cpp2::Value::Type::bool_value) {
+                    LOG(ERROR) << "Create Tag Failed: " << name
+                                << " type mismatch";
+                    return cpp2::ErrorCode::E_CONFLICT;
+                }
+                defaultValue = folly::to<std::string>(value->get_bool_value());
+                break;
+            case nebula::cpp2::SupportedType::INT:
+                if (value->getType() != nebula::cpp2::Value::Type::int_value) {
+                    LOG(ERROR) << "Create Tag Failed: " << name
+                                << " type mismatch";
+                    return cpp2::ErrorCode::E_CONFLICT;
+                }
+                defaultValue = folly::to<std::string>(value->get_int_value());
+                break;
+            case nebula::cpp2::SupportedType::DOUBLE:
+                if (value->getType() != nebula::cpp2::Value::Type::double_value) {
+                    LOG(ERROR) << "Create Tag Failed: " << name
+                                << " type mismatch";
+                    return  cpp2::ErrorCode::E_CONFLICT;
+                }
+                defaultValue = folly::to<std::string>(value->get_double_value());
+                break;
+            case nebula::cpp2::SupportedType::STRING:
+                if (value->getType() != nebula::cpp2::Value::Type::string_value) {
+                    LOG(ERROR) << "Create Tag Failed: " << name
+                                << " type mismatch";
+                    return cpp2::ErrorCode::E_CONFLICT;
+                }
+                defaultValue = folly::to<std::string>(value->get_string_value());
+                break;
+            case nebula::cpp2::SupportedType::TIMESTAMP:
+                if (value->getType() != nebula::cpp2::Value::Type::timestamp) {
+                    LOG(ERROR) << "Create Tag Failed: " << name
+                                << " type mismatch";
+                    return cpp2::ErrorCode::E_CONFLICT;
```

Comment:
why return `E_CONFLICT `

Reply:
Follow the previous.

Reply:
Previous may be an error, this data type or parameter is wrong

Reply:
Follow the behavior that create the tag/edge.